### DISCUSSION TITLE
Trigger prebuild programmatically

### DISCUSSION THREAD

**Discussion by alexswerner** [community/community#55760] [Codespaces]
### Select Topic Area

Question

### Body

Is there a way to trigger prebuilding a devcontainer for codespace usage using Actions or via the REST API?

**Comment by samruddhikhandale** [comment]
Hi 👋 

Currently, [prebuilding-your-codespaces/configuring-prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/configuring-prebuilds#configuring-prebuilds) (step 7) mentions few ways on how a prebuild can be automatically triggered. Also, one can manually trigger the prebuilds, see [managing-prebuilds#manually-trigger-prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/managing-prebuilds#manually-trigger-prebuilds)

@alexswerner  Have you looked at them? 

Unfortunately, we don't have a way to trigger a prebuild via actions or REST API. I am curious on the scenario you are trying to achieve here 👀

**Comment by alexswerner** [comment]
Hi @samruddhikhandale,

thanks for the reply. I have read the documentation, but what is mentioned in step7 does not work in my case.
Basically, what I do its to have my devcontainer be build off a custom docker image which contains a bunch of dependencies. The reduces build times, especially since I have multiple repos using the same base image. What i'm aiming for is

Repo1: contains Dockerfile for base image. This is build using an action. Runtime ~1h
Repo2: contains devcontainer with minimal additions to make the docker image a devcontainer. This i quickly being rebuild on every commit so any newly started codespace is getting the current state of the repo. Build time: 2min.
Repo3: same as Repo2

Now it would be nice if a rebuild of the base image triggers an update of the devcontainer.
Actually this is used in a classrooms.github setup. 

I'm asking for this just because it makes things more convenient. Like all actions usage. So having a REST API or actions integration would be great.

**Reply by samruddhikhandale** [reply]
Thanks for explaining the use case, sounds interesting and I'm sure others would benefit from programmatic triggers as well. 
I have opened an internal tracking issue for this request. 

In the meanwhile, how about using "Scheduled trigger"? I agree, that it doesn't 100% fit with your use case, however, if you have/create a new schedule for repo1 (ie for releasing images), then you can sync schedules for repo2 and repo3 as well (with some buffer).

**Reply by alexswerner** [reply]
Thanks for following up on this.

I considered Scheduled trigger, but the lack of determinism and consumption of cpu time is not so attractive. But thanks for the hint.

**Reply by mattbarreiro** [reply]
@samruddhikhandale are you able to share an ETA on this? 

A client of mine is looking at adopting Codespaces, but needs to use an internally-maintained container image as the base container image. This image needs to be built on-prem via a third-party build system for legacy reasons, and then gets pushed to GHCR.

While I could probably hack something together to force updates to prebuilds using tagged container images + Dependabot, it could  get difficult to maintain at scale. A much simpler solution it would be to have the image in `devcontainer.json` set to the `latest` tag, and add a step to the third-party build system to invoke prebuilds after the new image is uploaded.

**Comment by emilong** [comment]
I would also like this functionality - our use case is that we create new repositories for interview candidates in which they can use codespaces to do an exercise. We are currently manually configuring prebuilds on every single new repo, which is one more manual step we'd love to take out of our process. cc @samruddhikhandale
